src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,38 @@
 package mate.academy;
 
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final Map<EventListener, Boolean> listeners = new ConcurrentHashMap<>();
+    private final ExecutorService executorService = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        listeners.put(listener, true);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        listeners.keySet().forEach(listener -> CompletableFuture.runAsync(()
+                -> listener.onEvent(event), executorService));
     }
 
     public void shutdown() {
```

Comment:
The `shutdown` method attempts to gracefully shut down the executor service, but it also needs to handle the case where tasks are still running after `shutdownNow` is called. You might want to check if there are any tasks that did not terminate and take appropriate action, or log this condition.